ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -224,6 +224,9 @@ private BlobId(short version, BlobIdType type, byte datacenterId, short accountI
         this.accountId = accountId;
         this.containerId = containerId;
         this.isEncrypted = isEncrypted;
+        if (blobDataType == null) {
+          throw new IllegalArgumentException("blobDataType can't be null for id version " + BLOB_ID_V5);
+        }
         this.blobDataType = blobDataType;
```

Comment:
this is a tad better for readability
`this.blobDataType = Objects.requireNonNull(blobDataType, "blobDataType can't be null for id version " + BLOB_ID_V5);`

Reply:
Perhaps, but that utility throws NPE and IllegalArg is more precise IMO.  The same pattern is used for other parameters like partitionId.  I'd prefer to implement our own utility that throws the exception we want, and use that throughout.  I can take this in a follow up PR.